### PR TITLE
Issue 1602 verify signed messages(detached)

### LINKED ISSUE #1602: verify signed messages(detached)
it's similar to #1097

### PR DESCRIPTION
**Pull request by DenBond7**
This PR added verification for OpenPGP/MIME signed messages. unfortunately, I had to change the logic that relates to parsing data blocks.

close #1602

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/main/java/com/flowcrypt/email/core/msg/RawBlockParser.kt [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        return TreatAs.PLAIN_FILE
+      }
+    }
+  }
+
+  private fun checkForPublicKeyBlock(stream: InputStream): Boolean {
+    val a = ByteArray(100)
+    val r = stream.read(a)
+    if (r < 1) return false
+    val s = String(if (r == a.size) a else a.copyOf(r), StandardCharsets.US_ASCII)
```
Comment (IvanPizhenko) [inline FlowCrypt/src/main/java/com/flowcrypt/email/core/msg/RawBlockParser.kt:195]: a.size <= r

Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/core/msg/RawBlockParser.kt:196]: Are you sure? It's cut-paste from PgpMsg. I didn't change this code, just copied it.

**Comment by DenBond7** [comment]
@IvanPizhenko I've marked it as "ready to review". Don't worry about tests. it seems `Semaphore` changes CRLF -> CR when loading sources for tests. That's why new tests fail. While you doing a review I will think about fixing it.

File: FlowCrypt/src/main/java/com/flowcrypt/email/core/msg/RawBlockParser.kt (hunk 1 above)
Comment (IvanPizhenko) [reply FlowCrypt/src/main/java/com/flowcrypt/email/core/msg/RawBlockParser.kt:208]: Yes, I understand, but the above is a bit more optimal than what it is now.

**Review by IvanPizhenko — APPROVED** [review]

**Comment by IvanPizhenko** [comment]
@DenBond7 Looks good to me, just need to fix tests.

**Pull request merged**